src/parser.jl

```diff
@@ -2617,6 +2617,10 @@ function parse_array(ps::ParseState, mark, closer, end_is_symbol)
     # For an excellent overview of Pratt parsing, see
     # https://matklad.github.io/2020/04/13/simple-but-powerful-pratt-parsing.html
     (dim, binding_power) = parse_array_separator(ps, array_order)
+    if binding_power == typemin(Int)
+        bump_closing_token(ps, closer)
```

Comment:
Oh let's also add the case that can lead to this branch inline. It makes reading the code later a *lot* easier

```suggestion
        # [x@y  ==>  (hcat x (error-t ✘ y))
        bump_closing_token(ps, closer)
```

Reply:
My plan has kind of been to extract these inline comments automatically and use them as test cases ... haven't got around to that yet though

Reply:
Yeah, I saw that comment in the tests. Will be a bit annoying to keep the inline annotations in sync with the tests until that's done.